megamek/src/megamek/common/Entity.java

```diff
@@ -15071,6 +15080,491 @@ public String getTilesetModeString() {
         return "";
     }
     
+    //Tractors and trailers, tugs, etc
+    
+    /**
+     * Used to determine if this vehicle can be towed by a tractor
+     * 
+     * @return
+     */
+    public boolean isTrailer() {
+        return false;
+    }
+    
+    /**
+     * Used to determine if this vehicle can be the engine/tractor 
+     * for a bunch of trailers
+     * 
+     * @return
+     */
+    public boolean isTractor() {
+        return false;
+    }
+    
+    /**
+     * Returns a list of Coords that need to be checked for entities that can be towed
+     * This accounts for the hexes occupied by each entity in the 'train', plus hexes
+     * in front of or behind each trailer hitch 
+     * 
+     * @return
+     */
+    public ArrayList<Coords> getHitchLocations() {
+        ArrayList<Coords> trailerPos = new ArrayList<Coords>();
```

Comment:
When building a collection that can only have a single copy of a particular element, use a `Set` instead of a `List`. There is no need to check whether a `Set` already contains an element when adding it. The `HashSet` implementation requires that the contained type implements `equals(Object)` and `hashCode()`, which `Coords` does.

Reply:
Fixed